distributed/scheduler.py

```diff
@@ -4646,6 +4676,24 @@ def stimulus_task_finished(self, key=None, worker=None, stimulus_id=None, **kwar
                     "stimulus_id": stimulus_id,
                 }
             ]
+        elif ts.run_id != run_id:
```

Comment:
The logic here could likely be improved, but that would also mean some breaking changes to the transition logic and should be done in a PR focussing on that.